builtin/providers/aws/resource_aws_autoscaling_metrics_collection.go

```diff
@@ -0,0 +1,140 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsAutoscalingMetric() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAutoscalingMetricCreate,
+		Read:   resourceAwsAutoscalingMetricRead,
+		Update: resourceAwsAutoscalingMetricUpdate,
+		Delete: resourceAwsAutoscalingMetricDelete,
+
+		Schema: map[string]*schema.Schema{
+			"autoscaling_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"metrics": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+
+			"granularity": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAutoscalingMetricCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).autoscalingconn
+
+	props := &autoscaling.EnableMetricsCollectionInput{
+		AutoScalingGroupName: aws.String(d.Get("autoscaling_group_name").(string)),
+		Granularity:          aws.String(d.Get("granularity").(string)),
+	}
+
+	if v, ok := d.GetOk("metrics"); ok {
+		props.Metrics = expandStringList(v.(*schema.Set).List())
+	}
+
+	log.Printf("[INFO] Enabling metrics collection for the ASG: %s", d.Id())
+	_, err := conn.EnableMetricsCollection(props)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(d.Get("autoscaling_group_name").(string))
+
+	return resourceAwsAutoscalingMetricRead(d, meta)
+}
+
+func resourceAwsAutoscalingMetricRead(d *schema.ResourceData, meta interface{}) error {
+	g, err := getAwsAutoscalingGroup(d, meta)
+	if err != nil {
+		return err
+	}
+	if g == nil {
+		return nil
+	}
+
+	if g.EnabledMetrics != nil && len(g.EnabledMetrics) > 0 {
+		d.Set("metrics", flattenAsgEnabledMetrics(g.EnabledMetrics))
+		d.Set("granularity", g.EnabledMetrics[0].Granularity)
```

Comment:
I haven't looked too deep, but are we sure `EnabledMetrics[0]` is the correct one? Do we need to range over and match on something to be sure?